Library/Homebrew/utils.rb

```diff
@@ -770,3 +770,27 @@ def number_readable(number)
   (numstr.size - 3).step(1, -3) { |i| numstr.insert(i, ",") }
   numstr
 end
+
+# Truncates a text string to fit within a byte size constraint,
+# preserving character encoding validity. The returned string will
+# be no longer than the specified max_bytes, though the exact
+# shortfall may vary.
+def truncate_text_to_approximate_size(s, max_bytes, options = {})
+  front_weight = options[:front_weight] || 0.5
+  return s if s.bytesize <= max_bytes
+
+  if front_weight < 0 || front_weight > 1
+    raise "opts[:front_weight] must be between 0.0 and 1.0"
+  end
```

Comment:
Do we have a valid use case for setting this to 0 or 1? If not, I'd be more conservative and clamp this value to no less than 0.1 and no greater than 0.9. I'd be also tempted to *always* just clamp to this range if an invalid value is supplied instead of raising an exception here. (The exception can still be raised for `ARGV.homebrew_developer?`.)

Though to be honest, I'd prefer if we just hard-coded the value and only added the option back once a need arises.